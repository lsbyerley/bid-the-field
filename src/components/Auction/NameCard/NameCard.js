import { ScaleIcon } from '@heroicons/react/outline';
import useAsyncReference from '@/lib/useAsyncReference';

const NameCard = ({ auction }) => {
  const asyncAuction = useAsyncReference(auction, true);

  return (
    <div className='rounded-lg card compact bg-base-200'>
      <div className='justify-center card-body'>
        <div className='flex items-center'>
          <div className='flex-shrink-0'>
            <ScaleIcon className='w-6 h-6 text-gray-400' aria-hidden='true' />
          </div>
          <div className='flex-1 w-0 ml-5'>
            <dl>
              <dt className='text-sm font-medium truncate'>Auction</dt>
              <dd>
                <div className='font-medium'>{asyncAuction.current.name}</div>
              </dd>
            </dl>
          </div>
        </div>
      </div>
    </div>
  );
};

export default NameCard;
