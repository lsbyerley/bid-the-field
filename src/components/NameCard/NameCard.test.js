import React from 'react';
import { render, screen } from 'test-utils';
import NameCard from './NameCard';

const defaultProps = (overrides) => {
  return {
    auction: {
      name: 'test',
    },
    ...overrides,
  };
};

const renderNameCard = (props) => {
  const utils = render(<NameCard {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('NameCard', () => {
  it('should render and match snapshot', () => {
    const { home } = renderNameCard(defaultProps());
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });
});
