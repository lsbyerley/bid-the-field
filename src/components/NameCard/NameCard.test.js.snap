// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameCard should render and match snapshot 1`] = `
<div
  class="rounded-lg card compact bg-base-100"
>
  <div
    class="justify-center card-body"
  >
    <div
      class="flex items-center"
    >
      <div
        class="flex-shrink-0"
      >
        <svg
          aria-hidden="true"
          class="w-6 h-6 text-gray-400"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 6l3 1m0 0l-3 9a5.002 5.002 0 006.001 0M6 7l3 9M6 7l6-2m6 2l3-1m-3 1l-3 9a5.002 5.002 0 006.001 0M18 7l3 9m-3-9l-6-2m0-2v2m0 16V5m0 16H9m3 0h3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="flex-1 w-0 ml-5"
      >
        <dl>
          <dt
            class="text-sm font-medium truncate"
          >
            Auction
          </dt>
          <dd>
            <div
              class="font-medium"
            >
              test
            </div>
          </dd>
        </dl>
      </div>
    </div>
  </div>
</div>
`;
